e2e/add-kubernetes-to-workspace.sh

```diff
@@ -22,20 +22,26 @@ cd ${REPO_ROOT}
 cd ..
 WORKSPACE=$(pwd)
 
-# Set up go workspace to build with this version
 cd "${REPO_ROOT}"
 
-go work init
-
-go work use .
-go work use providers
+# We use to build with a go workspace; the problem is that this isn't the code in our repo and isn't the code in vendor.
```

Comment:
this feels hacky, how do we know that vendor contains what we need for the other modules if we're only doing this dynamically at runtime AND not copying go.work into the docker build?

Reply:
Yes, that's why we're turning it off here :-)

Reply:
I don't understand the comment though, a go workspace will use vendor and it makes sense to ...?

Reply:
I will send in a separate PR, it should be needed here

Reply:
Turns out it is needed to adding it back (go work changes go mod, causes docker build to fail)